packages/uploadthing/src/client.ts

```diff
@@ -304,3 +311,37 @@ export const generateClientDropzoneAccept = (fileTypes: string[]) => {
 
   return Object.fromEntries(mimeTypes.map((type) => [type, []]));
 };
+
+// Returns a full URL to the dev's uploadthing endpoint
+export function getFullApiUrl(maybeUrl?: string): URL {
+  const base = (() => {
+    if (typeof window !== "undefined") {
+      return window.location.origin;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/prefer-optional-chain
+    if (typeof process !== "undefined" && process?.env?.VERCEL_URL) {
+      return `https://${process.env.VERCEL_URL}`;
+    }
+
+    // @ts-expect-error - import meta is not defined in node
+    if (import.meta.env?.VERCEL_URL) {
+      // @ts-expect-error - import meta is not defined in node
+      return `https://${import.meta.env.VERCEL_URL}`;
+    }
+
+    return "http://localhost:3000";
+  })();
+
+  try {
+    const url = new URL(maybeUrl ?? "/api/uploadthing", base);
```

Comment:
is it possible for `maybeUrl` to be a full url (not just path)? Not sure what happens if you supply both a full url and `base` to the URL constructor.

Reply:
It looks like it ignores the `base` if you provide a non-relative url.

Reply:
yes if the url is absolute then it won't use the base